pkg/controller/controller_instance.go

```diff
@@ -99,20 +101,47 @@ const (
 	startingInstanceOrphanMitigationMessage string = "The instance provision call failed with an ambiguous error; attempting to deprovision the instance in order to mitigate an orphaned resource"
 
 	clusterIdentifierKey string = "clusterid"
+
+	minBrokerOperationRetryDelay time.Duration = time.Second * 1
+	maxBrokerOperationRetryDelay time.Duration = time.Minute * 20
 )
 
+type backoffEntry struct {
+	generation          int64
+	calculatedRetryTime time.Time // earliest time we should retry
+	dirty               bool      // true indicates new backoff should be calculated
```

Comment:
I find it cleaner to calculate the delay from start of request to start of request rather than from end of request to start of request. Asymptotically, it approaches the same delay in either case. I'm not convinced that it makes much difference operationally how the delay is calculated, so long as it is increasing. If the delay were calculated from start-to-start, then you would not need to keep track of a dirty bit. The `calculatedRetryTime` would be set directly in `setRetryBackoffRequired`.

Reply:
One issue I fought with here is that in one of the non-happy path loops, you provision, fail, then have to go through orphan mitigation which may fail multiple times, then you do the retry.  On short backoffs the prior calculated retry added no delay because orphan mitigation took "so long".  

I agree its a bit convoluted, but this will be a short term solution (measured in months) before its replaced with a re-design for proper backoff/retry.

Reply:
OK. That makes sense.